diff --git a/chrome/browser/ui/views/page_action/page_action_controller.cc b/chrome/browser/ui/views/page_action/page_action_controller.cc
index 41aa9c0770297d43d3e3ad40c228341fa83b2e83..b79e479de0f53edc233e9a79d053aa3d4bdffda0 100644
--- a/chrome/browser/ui/views/page_action/page_action_controller.cc
+++ b/chrome/browser/ui/views/page_action/page_action_controller.cc
@@ -73,6 +73,7 @@ void ScopedPageActionActivity::RegisterWillDestroyControllerCallback() {
   }
 }
 
+namespace chromium_impl {
 PageActionControllerImpl::PageActionControllerImpl(
     PinnedToolbarActionsModel* pinned_actions_model,
     PageActionModelFactory* page_action_model_factory,
@@ -414,7 +415,7 @@ std::unique_ptr<PageActionModelInterface> PageActionControllerImpl::CreateModel(
   if (page_action_model_factory_ != nullptr) {
     return page_action_model_factory_->Create(action_id, is_ephemeral);
   } else {
-    return std::make_unique<PageActionModel>(is_ephemeral);
+    return std::make_unique<::page_actions::PageActionModel>(is_ephemeral);
   }
 }
 
@@ -454,7 +455,7 @@ PageActionControllerImpl::CreatePageMetricsRecorder(
 }
 
 base::RepeatingCallback<void(PageActionTrigger)>
-PageActionControllerImpl::GetClickCallback(base::PassKey<PageActionView>,
+PageActionControllerImpl::GetClickCallback(base::PassKey<page_actions::PageActionView>,
                                            actions::ActionId action_id) {
   return base::BindRepeating(&PageActionControllerImpl::RecordClickMetric,
                              weak_factory_.GetWeakPtr(), action_id);
@@ -471,7 +472,7 @@ void PageActionControllerImpl::RecordClickMetric(
 
 base::RepeatingClosure
 PageActionControllerImpl::GetAnchoredMessageCloseCallback(
-    base::PassKey<PageActionView>,
+    base::PassKey<page_actions::PageActionView>,
     actions::ActionId action_id) {
   return base::BindRepeating(&PageActionControllerImpl::HideAnchoredMessage,
                              base::Unretained(this), action_id);
@@ -495,7 +496,7 @@ PageActionControllerImpl::RegisterOnWillDestroyCallback(
 }
 
 void PageActionControllerImpl::RegisterCallbacks(
-    base::PassKey<PageActionView>,
+    base::PassKey<page_actions::PageActionView>,
     actions::ActionId action_id,
     PageActionView* page_action_view) {
   CHECK(page_action_view);
@@ -513,6 +514,7 @@ void PageActionControllerImpl::OnIsChipShowingChanged(
   FindPageActionModel(action_id).SetIsChipShowing(PassKey(), is_chip_showing);
 }
 
+}  // namespace chromium_impl
 std::ostream& operator<<(std::ostream& os, const SuggestionChipConfig& config) {
   os << "{ should_animate: " << config.should_animate
      << ", should_announce_chip: " << config.should_announce_chip << " }";
